pkg/csi/cinder/controllerserver.go

```diff
@@ -357,8 +361,14 @@ func (cs *controllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateS
 		return nil, status.Error(codes.Internal, "Multiple snapshots reported by Cinder with same name")
 
 	} else {
+		// Add cluster ID to the snapshot metadata
+		properties := map[string]string{cinderCSIClusterIDKey: cs.Driver.cluster}
+		for mVal, mKey := range req.Parameters {
+			properties[mKey] = mVal
+		}
+
 		// TODO: Delegate the check to openstack itself and ignore the conflict
-		snap, err = cs.Cloud.CreateSnapshot(name, volumeId, &req.Parameters)
+		snap, err = cs.Cloud.CreateSnapshot(name, volumeId, &properties)
```

Comment:
this is not related to this PR
but seems CreateSnapshot and CreateVolume take req.Parameters in different manner

CreateVolume doesn't contain req.parameter into tags and in turn put into metadata...

https://github.com/container-storage-interface/spec/blob/master/spec.md
says param 
```
// Plugin specific parameters passed in as opaque key-value pairs.
  // This field is OPTIONAL. The Plugin is responsible for parsing and
  // validating these parameters. COs will treat these as opaque.
  // Use cases for opaque parameters:
  // - Specify a policy to automatically clean up the snapshot.
  // - Specify an expiration date for the snapshot.
  // - Specify whether the snapshot is readonly or read/write.
  // - Specify if the snapshot should be replicated to some place.
  // - Specify primary or secondary for replication systems that
  //   support snapshotting only on primary.
```

so we should re-consider this to avoid put all those param into metadata or vice versa (line 107 above) , make all param to volume req ..

Reply:
Yes, that's what I was going to ask... As you said, for volumes we [sanitize](https://github.com/kubernetes/cloud-provider-openstack/blob/master/pkg/csi/cinder/controllerserver.go#L107) req parameters, but for snapshots we put them all in metadata.
I didn't modify this behavior in my PR to support backward compatibility, but I agree that we need to reconsider what we actually put there.

Reply:
https://github.com/kubernetes/cloud-provider-openstack/issues/1545 opened